pfdl_scheduler/scheduler.py

```diff
@@ -443,11 +450,16 @@ def on_task_finished(self, task_api: TaskAPI) -> None:
         """Executes Scheduling logic when a Task is finished."""
         for callback in self.task_callbacks.task_finished:
             callback(task_api)
+
+        order_finished = False
         if task_api.task.name == "productionTask":
             self.running = False
+            order_finished = True
+            self.petri_net_logic.draw_petri_net()
+            self.notify(NotificationType.PETRI_NET, self.scheduler_id)
 
         log_entry = "Task " + task_api.task.name + " with UUID '" + task_api.uuid + "' finished."
```

Comment:
I am quite unhappy with the string concatenation, or at least make it "type-safe"